src/engine/input/key-mapping/InputGroupList.cs

```diff
@@ -0,0 +1,180 @@
+﻿using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using Godot;
+using Newtonsoft.Json;
+
+/// <summary>
+///   Used by OptionsMenu>Inputs>InputGroupContainer
+/// </summary>
+public class InputGroupList : VBoxContainer
+{
+    [Export]
+    public NodePath ConflictDialogPath;
+
+    internal static readonly PackedScene InputEventItemScene =
+        GD.Load<PackedScene>("res://src/engine/input/key-mapping/InputEventItem.tscn");
+
+    private static readonly Dictionary<string, List<InputEventWithModifiers>> DefaultControlScheme =
+        GetCurrentlyAppliedControlScheme();
+
+    private ConfirmationDialog conflictDialog;
+
+    private InputEventItem latestDialogCaller;
+    private InputEventItem latestDialogConflict;
+    private InputEventWithModifiers latestDialogNewEvent;
+
+    public InputGroupList()
+    {
+        Instance = this;
+    }
+
+    public delegate void ControlSchemeChangedDelegate(Dictionary<string, List<InputEventWithModifiers>> data);
+
+    public event ControlSchemeChangedDelegate OnControlSchemeChanged;
+
+    public static InputGroupList Instance { get; private set; }
+    public IEnumerable<InputGroupItem> AllGroupItems => GetChildren().OfType<InputGroupItem>();
+
+    public static Dictionary<string, List<InputEventWithModifiers>> GetDefaultControlScheme()
+    {
+        return DefaultControlScheme;
+    }
+
+    public static Dictionary<string, List<InputEventWithModifiers>> GetCurrentlyAppliedControlScheme()
+    {
+        return InputMap.GetActions().OfType<string>()
+            .ToDictionary(p => p,
+                p => InputMap.GetActionList(p).OfType<InputEventWithModifiers>()
+                    .ToList());
+    }
+
+    public Dictionary<string, List<InputEventWithModifiers>> GetCurrentlyPendingControlScheme()
+    {
+        return AllGroupItems.SelectMany(p => p.Actions)
+            .ToDictionary(p => p.InputName, p => p.Inputs.Select(x => x.AssociatedEvent).ToList());
+    }
+
+    public override void _Ready()
+    {
+        conflictDialog = GetNode<ConfirmationDialog>(ConflictDialogPath);
+    }
+
+    public void ShowDialog(InputEventItem caller, InputEventItem conflict, InputEventWithModifiers newEvent)
+    {
+        latestDialogCaller = caller;
+        latestDialogConflict = conflict;
+        latestDialogNewEvent = newEvent;
+
+        conflictDialog.DialogText = $"There is a conflict with {conflict.AssociatedAction.DisplayName}.\n" +
+            $"Do you want to remove the input from {conflict.AssociatedAction.DisplayName}?";
+        conflictDialog.PopupCenteredMinsize();
+    }
+
+    /// <summary>
+    ///   Get the input conflict if there are any
+    /// </summary>
+    /// <param name="item">The event with the new value</param>
+    /// <returns>The collision if any</returns>
+    public InputEventItem Conflicts(InputEventItem item)
```

Comment:
Would it be nice to get all the conflicting inputs?

Reply:
If there were two conflicts two separate popups would show up.

Reply:
How does that look? Do you mean that the popups would overlap each other and you can only see one? Or does it actually work correctly in that all the conflicts are visible?

Reply:
I mean that the game should ask you two seperate time in serial.
It notifies you about the first conflict. If the user wants to remove the conflicting input the second popup should appear.
I will rewrite the code to only show one popup